### ISSUE TITLE
on('error') in stream mode seems not work as the document said

### ISSUE THREAD

**Issue by yanxyz** [sindresorhus/got#97]
Example code:

```js
'use strict'

const got = require('got')
const url = 'https://github.com/xxx.jpg'

got(url, function (err) {
  console.log(err.statusCode) // 406
})

got(url)
.catch(function (err) {
  console.log(err.statusCode) // 406
})

got.stream(url)
.on('error', function (err) {
  console.log(err.statusCode) // output nothing, seams that there is no error event
})
```

**Comment by floatdrop** [comment]
Fixed in `4.2.0`

**Comment by yanxyz** [comment]
Thx
